db/migrate/20201111122823_backfill_updated_at_of_ended_subscriptions.rb

```diff
@@ -0,0 +1,12 @@
+class BackfillUpdatedAtOfEndedSubscriptions < ActiveRecord::Migration[6.0]
+  def change
+    ended_subscriptions = Subscription.where(
+      # The time the bug was deployed
+      "ended_at > ?", Time.zone.parse("2020-11-10 16:26:00 +0000")
```

Comment:
You could alternatively do:

```
Subscription.where("ended_at > updated_at")
```

Reply:
It looks like this gotcha has been a recurring them: https://github.com/alphagov/email-alert-api/search?q=update_all

We ought to fix it for the other occurrences: nullify and archive.

Reply:
There seems to be quite a few subscriptions (before the bug was introduced) where that query returns true.

E.g

```
irb(main):002:0>     ended_subscriptions = Subscription.where("ended_at > updated_at").count
=> 629854
irb(main):003:2* (    ended_subscriptions = Subscription.where(
irb(main):004:2*     # The time the bug was deployed
irb(main):005:2*     "ended_at > ?", Time.zone.parse("2020-11-10 16:26:00 +0000")
irb(main):006:0>   ).count )
irb(main):007:0*
=> 1663
```

Reply:
Yup - whenever someone did something with `update_all` on it. Arguably we don't _need_ to fix them, but we may as well.